src/transformers/trainer.py

```diff
@@ -398,53 +399,52 @@ def train(self, model_path: Optional[str] = None):
 
                     scheduler.step()
                     model.zero_grad()
-                    global_step += 1
+                    self.global_step += 1
+                    self.epoch = epoch + (step + 1) / len(epoch_iterator)
```

Comment:
We keep track of `epoch` so we can use it as an x-axis on comparative charts. Also it's calculated at each step so we have the right value in case we run `Trainer.evaluate` method outside of `Trainer.train` method.